File: exercise/C#/day22/EID/Sex.cs
using LanguageExt;

namespace EID
{
    public enum Sex
    {
        Sloubi = 1,
        Gagna = 2,
        Catact = 3
    }

    public static class SexParser
    {
        public static Either<ParsingError, Sex> Parse(char potentialSex)
            => potentialSex switch
            {
                '1' => Sex.Sloubi,
                '2' => Sex.Gagna,
                '3' => Sex.Catact,
                _ => new ParsingError("Not a valid sex")
            };
    }
}

File: exercise/scala/day22/src/main/scala/eid/Sex.scala
package eid

enum Sex(val symbol: Char) {
  case Sloubi extends Sex('1')
  case Gagna extends Sex('2')
  case Catact extends Sex('3')

  override def toString: String = symbol.toString
}

object Sex {
  def parse(potentialSex: Char): Either[ParsingError, Sex] = Sex.values
    .find(_.symbol == potentialSex)
    .toRight(ParsingError("Not a valid sex"))
}